R/utils.R

```diff
@@ -291,6 +291,20 @@ check_sampler_transitions_treedepth <- function(post_warmup_sampler_diagnostics,
   }
 }
 
+check_bfmi <- function(post_warmup_sampler_diagnostics) {
+  if (!is.null(post_warmup_sampler_diagnostics)) {
+    energy <- posterior::extract_variable_matrix(post_warmup_sampler_diagnostics, "energy__")
+    ebfmi <- apply(energy, 2, function(x) {
+      (sum(diff(x)^2)/length(x))/var(x)
+    })
+    if (any(ebfmi < .3)) {
+      message(sum(ebfmi < .3), " of ", length(ebfmi) , " chains had estimated Bayesian fraction
+      of missing information(E-BFMI) less than 0.3, which may indicate poor exploration of the 
+      posterior. Try to reparameterize the model.")
```

Comment:
Thinking about this a bit more, I think we should remove "Try to reparameterize the model". At least for divergences and max treedepth we have a few concrete recommendations, but I think this is too vague of a suggestion to be helpful to a user. And even for divergences and treedepth I think we should move in the direction of removing the recommended solutions from the warning messages themselves and having the warning messages just be warnings with a pointer to resources that provide recommendations (that way we can also have the recommendations evolve without having to keep changing the warning messages).